### PR TITLE
Trace identity

### PR DESCRIPTION
**Pull request by SuyogSoti**
add decorators and policy

For reviewers, there are two things to look for.

1. I have added the DistributedTracingPolicy to the correct place.
1. I propagate the context with `tracing_context.with_current_context` any time a new thread is created. This should work with both `threading` and `ThreadPoolExecutor`. An example is the following:

`future = executor.submit(tracing_context.with_current_context(uploader.process_chunk), chunk)`

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/azure/identity/aio/_authn_client.py [hunk 1]
```diff
@@ -19,28 +20,33 @@ class AsyncAuthnClient(AuthnClientBase):
     """Async authentication client"""
 
     def __init__(
-        self,
-        auth_url: str,
-        config: Optional[Configuration] = None,
-        policies: Optional[Iterable[HTTPPolicy]] = None,
-        transport: Optional[AsyncHttpTransport] = None,
-        **kwargs: Mapping[str, Any]
+            self,
```
[the remark this answers is not in the document]
Comment (SuyogSoti) [reply sdk/identity/azure-identity/azure/identity/aio/_authn_client.py]: pylint wants me to...

File: sdk/identity/azure-identity/azure/identity/aio/_authn_client.py [hunk 2]
```diff
@@ -19,28 +20,33 @@ class AsyncAuthnClient(AuthnClientBase):
...
+            auth_url: str,
+            config: Optional[Configuration] = None,
+            policies: Optional[Iterable[HTTPPolicy]] = None,
+            transport: Optional[AsyncHttpTransport] = None,
+            **kwargs: Mapping[str, Any]
     ) -> None:
         config = config or self.create_config(**kwargs)
-        policies = policies or [ContentDecodePolicy(), config.logging_policy, config.retry_policy]
+        policies = policies or [
```
[the remark this answers is not in the document]
Comment (SuyogSoti) [reply sdk/identity/azure-identity/azure/identity/aio/_authn_client.py]: done

File: sdk/identity/azure-identity/azure/identity/credentials.py [hunk 3]
```diff
@@ -176,6 +180,7 @@ def create_config(**kwargs):
         """
         return Configuration(**kwargs)
 
+    @distributed_trace
     def get_token(self, *scopes):
```
[the remark this answers is not in the document]
Comment (SuyogSoti) [reply sdk/identity/azure-identity/azure/identity/credentials.py]: done

File: sdk/identity/azure-identity/azure/identity/_authn_client.py [hunk 4]
```diff
@@ -73,6 +74,10 @@ def _deserialize_and_cache_token(self, response, scopes, request_time):
 
         # now we have an int expires_on, ensure the cache entry gets it
         payload["expires_on"] = expires_on
+        if "expires_in" in payload:
```
[the remark this answers is not in the document]
Comment (SuyogSoti) [reply sdk/identity/azure-identity/azure/identity/_authn_client.py]: I have the version from the dev_setup.py -> `msal==0.3.1`.

should I get rid of it? the casting?

[the remark this answers is not in the document]
Comment (SuyogSoti) [reply sdk/identity/azure-identity/azure/identity/_authn_client.py]: done

File: sdk/identity/azure-identity/azure/identity/aio/_authn_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/aio/_authn_client.py]: We get our indentation from black and have enough already, thanks 😎

**Comment by SuyogSoti** [comment]
@johanste says not spans should be created for get_token here

**Pull request merged**